js/build/field-attachment.js

```diff
@@ -213,8 +213,21 @@ var ShortcodeAttribute = Backbone.Model.extend({
 		label:       '',
 		type:        '',
 		value:       '',
-		placeholder: '',
+		description: '',
+		customAttributes: {
+			placeholder: '',
+		}
 	},
+
+	initialize: function() {
+
+		// Handle legacy custom attributes.
+		if ( this.get('placeholder' ) ) {
```

Comment:
Can we have this compatibility shim just defined in one place, and file an issue to remove it in v0.4?